File: SoniApp/Models/Message.swift
//
//  Message.swift
//  SoniApp
//

import Foundation

/// Network DTO for messages.
struct Message: Identifiable, Codable {
    let id: String
    let text: String
    let senderId: String
    let receiverId: String
    let date: String?
    let senderName: String?
    let isRead: Bool?
    let readAt: String?
    let imageUrl: String?
    let clientId: String?
    
    enum CodingKeys: String, CodingKey {
        case id = "_id"
        case text
        case senderId
        case receiverId
        case date
        case senderName
        case isRead
        case readAt
        case clientId
        case imageUrl
    }
}

File: nodejsServer/models/Message.js
const mongoose = require('mongoose');

const MessageSchema = new mongoose.Schema({
  text: { type: String, required: false },
  senderId: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  receiverId: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  date: { type: Date, default: Date.now },
  senderName: { type: String, default: '' },
  isRead: { type: Boolean, default: false },
  readAt: { type: Date, default: null },
  clientId: { type: String, default: null },
  imageUrl: String,

});

module.exports = mongoose.model('Message', MessageSchema);
